### DISCUSSION TITLE
Nuxt3 Auth Middleware direction?

### DISCUSSION THREAD

**Discussion by dbhagen** [nuxt/nuxt#18177] [Questions]
Looking to write something simple to check state for auth and redirect if needed. Trying to parse through the output from `req` and `res`:
```
import type { IncomingMessage, ServerResponse } from 'http'
export default async (req: IncomingMessage, res: ServerResponse) => {
  console.log('req', req)
}
```

But it's a slow go. Does anyone have pointers on where to dig?

**Comment by crotanite** [comment]
I've been working on something similar for auth with Nuxt, but instead of using the server middleware I added a plugin that has allowed me to directly hook into the router:

```
import { defineNuxtPlugin } from '#app'

export default defineNuxtPlugin((nuxtApp: any) => {
	const router = nuxtApp.$router

	router.beforeEach((to: object, from: object, next: Function) => {
		console.log('beforeEach')
		next()
	})
})
```

**Reply by dbhagen** [reply]
Ok, I was wondering if it needs to be a plug-in instead. I like that I can clearly scope the two ends of it with the `auth.server.ts` vs. `auth.client.ts`.

Thanks for the pointer!
